### PR TITLE
Class constant annotation addition

### PR DESCRIPTION
**Pull request by matthieu88160**
Draft for #408

Adding Target::TARGET_CONSTANT
Adding getConstantAnnotations method to Reader
Adding getConstantAnnotation method to Reader

### CODE REVIEW

File: lib/Doctrine/Common/Annotations/AnnotationReader.php [hunk 1]
```diff
@@ -386,4 +419,27 @@ private function collectParsingMetadata($reflection): void
 
         $this->ignoredAnnotationNames[$type][$name] = $ignoredAnnotationNames;
     }
+
+    /**
+     * Retrieves imports for constants.
+     *
+     * @return array<string, class-string>
+     */
+    private function getConstantImports(ReflectionClassConstant $constant)
```
Comment (greg0ire) [inline lib/Doctrine/Common/Annotations/AnnotationReader.php]:
```suggestion
    private function getConstantImports(ReflectionClassConstant $constant): array
```

File: lib/Doctrine/Common/Annotations/AnnotationReader.php [hunk 2]
```diff
@@ -386,4 +419,27 @@ private function collectParsingMetadata($reflection): void
 
         $this->ignoredAnnotationNames[$type][$name] = $ignoredAnnotationNames;
     }
+
+    /**
+     * Retrieves imports for constants.
+     *
+     * @return array<string, class-string>
```
Comment (greg0ire) [inline lib/Doctrine/Common/Annotations/AnnotationReader.php]:
```suggestion
     * @return array<string, class-string|trait-string>
```

**Review by greg0ire — COMMENTED** [review]
This should target 1.14.x since it's a new feature

File: lib/Doctrine/Common/Annotations/Reader.php [hunk 3]
```diff
@@ -77,4 +78,26 @@ public function getPropertyAnnotations(ReflectionProperty $property);
...
+
+    /**
+     * Gets the annotations applied to a constant.
+     *
+     * @param ReflectionClassConstant $constant The ReflectionClassConstant of the constant
+     *                                          from which the annotations should be read.
+     *
+     * @return array<object> An array of Annotations.
+     */
+    public function getConstantAnnotations(ReflectionClassConstant $constant);
```
Comment (stof) [inline lib/Doctrine/Common/Annotations/Reader.php:90]: Adding new methods in an interface is a BC break

**Comment by derrabus** [comment]
Closing because we won't add new features to this library.

**Pull request closed without merging**